Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -616,40 +648,45 @@ namespace Terrain
                     lodIndex2 += LodGridVerts1D;
                 }
 
-                if (lodHeights[lodIndex1] == NoTerrainVertexHeight || lodHeights[lodIndex2] == NoTerrainVertexHeight)
+                uint16_t zIndex = m_vertexOrder.at(index);
```

Comment:
Nits:
1. Could be const.
2. Maybe call it "interleavedIndex"? Using "zIndex" right next to "xPos" and "yPos" really makes it look like it's referring to a z coordinate.

Reply:
interleavedIndex is less descriptive though - it just happens to reference how we constructed the index, not what the index actually is.. Maybe zOrderIndex? I know it's confusing this stuff is called z order, but that's how it's most commonly referred to so I don't want to come up with some other name for it.